File: KthLargestElementInAnArray/KthLargestElementInAnArray.py
class Solution(object):
    def findKthLargest(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        L = 0
        R = len(nums) - 1
        while L < R:
            left = L
            right = R
            key = nums[left]
            while left < right:
                while nums[right] < key and left < right:
                    right -= 1
                nums[left] = nums[right]
                while nums[left] >= key and left < right:
                    left += 1
                nums[right] = nums[left]
            nums[left] = key
            if left == k - 1:
                return nums[k - 1]
            elif left > k - 1:
                R = left - 1
            else:
                L = left + 1
        return nums[k - 1]

File: KthLargestElementInAnArray/KthLargestElementInAnArray.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        int left = 0, right = nums.size() - 1;
        while(left < right) {
        	int L = left, R = right;
        	int key = nums[L];
        	while(L < R) {
        		while(nums[R] < key && L < R)
        			R--;
        		nums[L] = nums[R];
        		while(nums[L] >= key && L < R)
        			L++;
        		nums[R] = nums[L];
        	}
        	nums[L] = key;
        	if(L == k - 1)
        		return nums[k - 1];
        	else if(L > k - 1)
        		right = L - 1;
        	else
        		left = L + 1;
        }
        return nums[k - 1];
    }
};